### ISSUE TITLE
 Indentation rule fails on every line after inline lambda in chain

### ISSUE THREAD

**Issue by FWDekker** [pinterest/ktlint#1320]
## Expected Behavior
ktlint indicates that the file has no formatting errors.

## Observed Behavior
ktlint indicates that every line is missing one level of indentation after a line that has an inline lambda function that is not the last argument to a function, and that function is then chained.

## Steps to Reproduce
With the following code sample,
```kotlin
package com.fwdekker.lintexample

class LintExample {
    fun doWithLambda(a: () -> Int, b: Int): Int {
        return 1
    }

    fun main() {
        doWithLambda({ 4 }, 5).toDouble()
    }
}
```
run `./gradlew ktlintCheck` from [ktlint-gradle](https://github.com/JLLeitschuh/ktlint-gradle) to get the following errors:
```
[redacted]/LintExample.kt:10:1 Unexpected indentation (4) (should be 8)
[redacted]/LintExample.kt:11:1 Unexpected indentation (0) (should be 4)
[redacted]/LintExample.kt:12:1 Unexpected indentation (0) (should be 4)
```

### Mitigations
The following changes to the code sample circumvent the issue:
* Store the output of `doWithLambda` in a variable, and then call `toDouble` on that on a separate line.
* Swap the arguments of `doWithLambda` so that the lambda is the last argument. It does not matter if the lambda is in the parentheses or not.
* Store the lambda in a separate variable and then pass it to `doWithLambda`.
* Reformat the invocation to:
  ```kotlin
  doWithLambda(
      {
          4
      },
      5
  ).toDouble()
  ```

Putting a `return@doWithLambda` inside the lambda does not resolve the issue.

## Your Environment
* ktlint: 0.42.1
* Integration: [Detekt](https://github.com/detekt/detekt) v1.19.0, or [ktlint-gradle](https://github.com/JLLeitschuh/ktlint-gradle) v10.2.0
* Gradle: v7.3.2
* OS: Debian 11
* Project: The error occurs on [line 114 of `RandomnessIcons.kt`](https://github.com/FWDekker/intellij-randomness/blob/5ba44c067a6e8d4d7c90c9f4265701f9d43e06bf/src/main/kotlin/com/fwdekker/randomness/RandomnessIcons.kt#L114); you have to manually change the Detekt version from 1.18.1 to 1.19.0

**Comment by paul-dingemans** [comment]
> ktlint: I don't know how to check this :-(

run `ktlint --version`

**Comment by FWDekker** [comment]
`Unknown command: ktlint`

I'm using the [ktlint-gradle wrapper](https://github.com/JLLeitschuh/ktlint-gradle) to run ktlint. I checked their changelogs instead and I think the default ktlint version is 0.42.1.

**Comment by MeikeMertschFortum** [comment]
I have the same issue. [I reported it first to the lib I use for the gradle implementation](https://github.com/JLLeitschuh/ktlint-gradle/issues/565), but I was sent here.

Since that library updated the default ktlint version from `0.41.0` to `0.42.1`, I get surprising complaints about the indentation in one of my classes:

Here's the class code:

```kotlin
class Mailbox(private val topic: String) {

    companion object {
        fun fromMailAddress(mail: String) = Mailbox(getTopicFrom(mail))
    }

    fun expectSubscriptionMailIn(waitingTimeInSeconds: Int): Email =
        waitUntil({ it.isSubConfirmation() }, "Subscription mail for $topic", waitingTimeInSeconds).first()

    fun expectSessionReportIn(waitingTimeInSeconds: Int): String =
        waitUntil({ it.isSessionReport() }, "Session Report for $topic", waitingTimeInSeconds).first()
            .returnReportLink()

    fun expectErrorMailIn(waitingTimeInSeconds: Int, errorCode: String): Email =
        waitUntil({ it.isErrorMessage(errorCode) }, "Error mail for $topic", waitingTimeInSeconds).first()

    fun expectOfflineMailIn(waitingTimeInSeconds: Int): Email =
        waitUntil({ it.isOfflineError() }, "Offline mail for $topic", waitingTimeInSeconds).first()

    fun expectFleetInvitationIn(waitingTimeInSeconds: Int): String =
        waitUntil({ it.isFleetInvite() }, "Fleet invite for $topic", waitingTimeInSeconds).first()
            .findLink()

    fun expectMailsBySubject(vararg subjectCandidates: String) =
        waitUntil({ subjectCandidates.contains(it.subject) }, "Mail for $topic", 60)

    fun subscribeToAwsNotifications(): Mailbox = apply {
        val subscriptionEmail = expectSubscriptionMailIn(90)
        subscriptionEmail.subscribe()
    }

    private fun waitUntil(predicate: Predicate<Email>, alias: String, waitingTimeInSeconds: Int): List<Email> =
        checkMailsFor(predicate, alias, waitingTimeInSeconds, topic)
}
```

now the check failed as follows:
```
> Task :ktlintTestSourceSetCheck FAILED
.../Mailbox.kt:19:1 Unexpected indentation (4) (should be 8)
.../Mailbox.kt:20:1 Unexpected indentation (8) (should be 12)
.../Mailbox.kt:21:1 Unexpected indentation (12) (should be 20)
.../Mailbox.kt:23:1 Unexpected indentation (4) (should be 12)
.../Mailbox.kt:24:1 Unexpected indentation (8) (should be 16)
.../Mailbox.kt:26:1 Unexpected indentation (4) (should be 16)
.../Mailbox.kt:27:1 Unexpected indentation (8) (should be 20)
.../Mailbox.kt:29:1 Unexpected indentation (4) (should be 20)
.../Mailbox.kt:30:1 Unexpected indentation (8) (should be 24)
.../Mailbox.kt:31:1 Unexpected indentation (12) (should be 32)
.../Mailbox.kt:33:1 Unexpected indentation (4) (should be 24)
.../Mailbox.kt:34:1 Unexpected indentation (8) (should be 28)
.../Mailbox.kt:36:1 Unexpected indentation (4) (should be 24)
```

if I run the formatter, I end up with this:

```kotlin
class Mailbox(private val topic: String) {

    companion object {
        fun fromMailAddress(mail: String) = Mailbox(getTopicFrom(mail))
    }

    fun expectSubscriptionMailIn(waitingTimeInSeconds: Int): Email =
        waitUntil({ it.isSubConfirmation() }, "Subscription mail for $topic", waitingTimeInSeconds).first()

        fun expectSessionReportIn(waitingTimeInSeconds: Int): String =
            waitUntil({ it.isSessionReport() }, "Session Report for $topic", waitingTimeInSeconds).first()
                    .returnReportLink()

            fun expectErrorMailIn(waitingTimeInSeconds: Int, errorCode: String): Email =
                waitUntil({ it.isErrorMessage(errorCode) }, "Error mail for $topic", waitingTimeInSeconds).first()

                fun expectOfflineMailIn(waitingTimeInSeconds: Int): Email =
                    waitUntil({ it.isOfflineError() }, "Offline mail for $topic", waitingTimeInSeconds).first()

                    fun expectFleetInvitationIn(waitingTimeInSeconds: Int): String =
                        waitUntil({ it.isFleetInvite() }, "Fleet invite for $topic", waitingTimeInSeconds).first()
                                .findLink()

                        fun expectMailsBySubject(vararg subjectCandidates: String) =
                            waitUntil({ subjectCandidates.contains(it.subject) }, "Mail for $topic", 60)

                        fun subscribeToAwsNotifications(): Mailbox = apply {
                            val subscriptionEmail = expectSubscriptionMailIn(90)
                            subscriptionEmail.subscribe()
                        }

                        private fun waitUntil(predicate: Predicate<Email>, alias: String, waitingTimeInSeconds: Int): List<Email> =
                            checkMailsFor(predicate, alias, waitingTimeInSeconds, topic)
                    }
```

I don't really understand why. It feels like the equal signs work almost as opening parentheses but not closing ones.
Even more curious is that this is the only file in the project that has this issue (but I am almost certain it is also the only one that contains only single expression functions that are longer than a single line (with exception of the companion object)

And I am pretty convinced that this format is not how it should be. Ktlint usually gives me beautiful code (yay!), but this doesn't look right 😅

**Comment by paul-dingemans** [comment]
@MeikeMertschFortum Your problem will be solved with https://github.com/pinterest/ktlint/pull/1284

**Comment by paul-dingemans** [comment]
Closed by #1284
